src/mds/MDSMap.cc

```diff
@@ -551,6 +553,7 @@ void MDSMap::encode(bufferlist& bl, uint64_t features) const
   ::encode(session_autoclose, bl);
   ::encode(max_file_size, bl);
   ::encode(max_mds, bl);
+  ::encode(balancer, bl);
```

Comment:
When you add an item to an encoded representation, you have to add it at the end (i.e. after "damaged" in this case) -- the way older receivers handle newer messages is to ignore trailing bytes